File: test/hello_core/test_greeter.cc
#include <hello_core/greeter.hh>

#include <catch2/catch_test_macros.hpp>

TEST_CASE("Greeter produces the expected greeting", "[hello_core][greeter]") {
  const hello_core::Greeter greeter("World");
  REQUIRE(greeter.greet() == "Hello, World!");
}

File: src/hello_core/python/tests/test_greeter.py
from hello_core import Greeter, HelloCoreError


def test_greet_basic() -> None:
    with Greeter("World") as greeter:
        assert greeter.greet() == "Hello, World!"


def test_greeter_reusable_after_close() -> None:
    greeter = Greeter("Ada")
    assert greeter.greet() == "Hello, Ada!"
    greeter.close()
    greeter.close()  # double-close must be a no-op, not a crash


def test_error_type_importable() -> None:
    assert issubclass(HelloCoreError, RuntimeError)
